### PR TITLE
Add AutoConfigurator (former HP Tool)

### PR DESCRIPTION
**Pull request by markelsanz14**
# What does this PR do ?
Adds AutoConfigurator to NeMo Megatron directory.

**Collection**: nemo_megatron_launcher

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [X] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?
@okuchaiev @yaoyu-33 @ericharper @Davood-M 

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ericharper** [comment]
Can you fix the style with `python setup.py style --fix` ?

**Comment by ericharper** [comment]
Please also look into the Code scanning results.

File: nemo_megatron_launcher/auto_configurator/autoconfig/training_config.py [hunk 1]
```diff
@@ -0,0 +1,814 @@
+"""Prepares and launches the training HP search using bignlp-scripts."""
```
Comment (ericharper) [inline nemo_megatron_launcher/auto_configurator/autoconfig/training_config.py:1]: can change bignlp -> nemo megatron launcher

**Review by ericharper — CHANGES_REQUESTED** [review]
Please see the code scanning alerts and a few files are still missing copyrights.

**Review by titu1994 — CHANGES_REQUESTED** [review]
This must go to a NeMo tool repo.

Refer to https://github.com/NVIDIA/NeMo/pull/5657#pullrequestreview-1224968295

**Review by dimapihtar — CHANGES_REQUESTED** [review]
Plaese check @ericharper comments about code scanning.

**Comment by markelsanz14** [comment]
I think this PR is ready. Please confirm if ready @ericharper @okuchaiev

**Pull request closed without merging**